pex/pex_builder.py

```diff
@@ -710,15 +764,20 @@ def _build_packedapp(
     ):
         # type: (...) -> None
 
-        pex_info = self._pex_info.copy()
-        pex_info.update(PexInfo.from_env())
+        pex_info = self._setup_pex_info()
 
         # Include user sources, PEX-INFO and __main__ as loose files in src/.
-        for fileset in ("executable", "importhook", "main", "manifest", "resource", "source"):
-            for f in self._chroot.filesets.get(fileset, ()):
-                dest = os.path.join(dirname, f)
-                safe_mkdir(os.path.dirname(dest))
-                safe_copy(os.path.realpath(os.path.join(self._chroot.chroot, f)), dest)
+        with TRACER.timed("copying over uncached sources", V=9):
```

Comment:
Ok, this seems fine - but boy oh boy does it seem _incredibly_ complicated vs just ordering the labels correctly in the existing tuple on the LHS and writing a good block comment just above.

Reply:
You're right, and this change could be reduced quite a bit by providing the labels literally (for example, we could avoid changing this method at all). I thought it was useful to document the meanings of each label, since they're not really documented elsewhere unless you grep for the literal strings `"executable"`, `"importhook"`, etc. I was also thinking that naming each section with a classmethod would make it easier for someone to subclass `PEXBuilder` and insert their own files into the appropriate section while retaining the performance benefit for zipapps. If that's not a supported use case, we could very well reduce the size of this change by a lot.